hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java

```diff
@@ -270,18 +270,15 @@ public void verifyAndFixupContainerData(ContainerData containerData)
    */
   private boolean resolveDuplicate(KeyValueContainer existing,
       KeyValueContainer toAdd) throws IOException {
-    if (existing.getContainerData().getReplicaIndex() != 0 ||
-        toAdd.getContainerData().getReplicaIndex() != 0) {
-      // This is an EC container. As EC Containers don't have a BSCID, we can't
-      // know which one has the most recent data. Additionally, it is possible
-      // for both copies to have a different replica index for the same
-      // container. Therefore we just let whatever one is loaded first win AND
-      // leave the other one on disk.
-      LOG.warn("Container {} is present at {} and at {}. Both are EC " +
-              "containers. Leaving both containers on disk.",
-          existing.getContainerData().getContainerID(),
-          existing.getContainerData().getContainerPath(),
-          toAdd.getContainerData().getContainerPath());
+    if (existing.getContainerData().getReplicaIndex() != 0 &&
+        existing.getContainerData().getReplicaIndex() != toAdd.getContainerData().getReplicaIndex()) {
```

Comment:
Even if it is equal we have not swapped here right? If 2 replica indices are equal do we want to remove one replica only if the block files are same? Given that this is a delete operation IMO we should be a bit careful

Reply:
Its being swapped based on State as down logic after if, for index same,
- if existing is closed, given preference to this
- else if new is closed, given preference to this
- else preference is given to existing which is already loaded.